### ISSUE TITLE
pip3 broken on emsdk python3 on macOS

### ISSUE THREAD

**Issue by juj** [emscripten-core/emsdk#704]
Attempting to install `sphinx-build` via pip3 on macOS gives

```
MacBook-Pro:master clb$ pip3
-bash: /Users/clb/emsdk/python/3.7.4-2_64bit/bin/pip3: /usr/local/bin/python3.7: bad interpreter: No such file or directory
```

**Comment by sbc100** [comment]
This makes me think.. we probably should not be putting our version of python (or pip3 in the user's PATH).   The version of python we supply in the emsdk should probably be for internal use only.  

Otherwise we potentially mess up the python installation that the developer might have done on their own.   Our entry points for emsdk, emcc, etc.. should all be able to find python via EMSDK_PYTHON, so adding it the PATH should not be necessary right?  At least we shouldn't do it by default.   Same for node I think.   Most developers probably have their own version they like

**Comment by juj** [comment]
That rationale does make some sense, although one can also argue the opposite making sense: if I'm inside emsdk_env cmdline environment, I might want to use specifically the emsdk supplied tools (as long as they worked). It is super convenient to be able to write

```
emcc tests/hello_world.c -o hello_world.js
node hello_world.js
```

on all OSes without needing to do extra manual steps after having entered `emsdk_env`.

**Comment by sbc100** [comment]
Agreed, although we should at least add this as an option.  

We could even make it somewhat automatic. i.e. only add node and python to the `PATH` if the user doesn't already have a copy of node or python in their PATH.   Either way, its good that neither emsdk nor emscripten *depend* on node or python being in the PATH at all (python is found via `$EMSDK_PYTHON` and node is found via `NODE` in emscripten config).   This allows emsdk to be somewhat hermetic.

So adding this stuff to the `PATH` is a user convenience feature and not an essential feature.    We could make it semi-automatic by only added node/python to the PATH if the user doesn't already have them in their PATH?

**Comment by juj** [comment]
> Either way, its good that neither emsdk nor emscripten _depend_ on node or python being in the PATH at all

Agreed, that has been one of the design decisions since the start: Emscripten itself should only be looking at EM_CONFIG, and not need a family of env. vars or PATH setup - all that is for the benefit of the end user of emsdk.

That also helps if people are programmatically setting up CI or other environments for emscripten - they do not need to necessarily mimic all the fields that emsdk sets, but can just set that one config env. var and fill out the emscripten config file to do the appropriate setup.

**Comment by sbc100** [comment]
Whatever else we do we should probably *also* fix the pip3 script so that its is not tied to a specific path like that. (So keeping this open for now)

**Comment by sbc100** [comment]
Putting actually fixing pip3 on the back burning now that we have mitigated the effect.   Hopefully this is no longer urgent.

**Comment by sbc100** [comment]
Closing since the workaround in #706 seems to be enough here.
